ansible/configs/three-tier-app/pre_software.yml

```diff
@@ -8,21 +9,61 @@
   roles:
     - { role: "set-repositories", when: 'repo_method is defined' }
     - { role: "common", when: 'install_common' }
-    - { role: "set_env_authorized_key", when: 'set_env_authorized_key' }
+    - { role: "set_env_authorized_key", when: set_env_authorized_key | bool }
   tags:
     - step004
     - common_tasks
 
 - name: Configuring Bastion Hosts
   hosts: bastions
   become: true
-  roles:
-    -  { role: "bastion", when: 'install_bastion' }
-    - role: bastion-student-user
-      when: install_student_user | bool
   tags:
     - step004
     - bastion_tasks
+  roles:
+    -  { role: "bastion", when: install_bastion | bool }
+    -  { role: bastion-student-user, when: install_student_user | bool }
```

Comment:
i would use `include_role` when possible

as a task, not in the play definition

Reply:
Yes, You are right but I thought not to change any thing which is already exist instead add new stuff.  Any way will change to include_role.

Reply:
@prakhar1985 @fridim , I will change as per suggestion and push to git

Reply:
@fridim and @prakhar1985 I have updated the code as per suggestion.